presto-hudi/src/main/java/com/facebook/presto/hudi/HudiPageSource.java

```diff
@@ -0,0 +1,243 @@
+                }
+                else {
+                    blocks[i] = dataPage.getBlock(delegateIndexes[i]);
+                }
+            }
+            return new Page(batchSize, blocks);
+        }
+        catch (RuntimeException e) {
+            closeWithSuppression(e);
+            throwIfInstanceOf(e, PrestoException.class);
+            throw new PrestoException(HUDI_CURSOR_ERROR, e);
+        }
+    }
+
+    @Override
+    public void close()
+    {
+        try {
+            delegate.close();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public String toString()
+    {
+        return delegate.toString();
+    }
+
+    @Override
+    public long getSystemMemoryUsage()
+    {
+        return delegate.getSystemMemoryUsage();
+    }
+
+    protected void closeWithSuppression(Throwable throwable)
+    {
+        requireNonNull(throwable, "throwable is null");
+        try {
+            close();
+        }
+        catch (RuntimeException e) {
+            // Self-suppression not permitted
+            if (throwable != e) {
+                throwable.addSuppressed(e);
+            }
+        }
+    }
+
+    private static Object deserializePartitionValue(Type type, String valueString, String name, TimeZoneKey timeZoneKey)
+    {
+        if (valueString == null) {
+            return null;
+        }
+
+        try {
+            if (type.equals(BOOLEAN)) {
+                if (valueString.equalsIgnoreCase("true")) {
+                    return true;
+                }
+                if (valueString.equalsIgnoreCase("false")) {
+                    return false;
+                }
+                throw new IllegalArgumentException();
+            }
+            if (type.equals(INTEGER)) {
+                return parseLong(valueString);
+            }
+            if (type.equals(BIGINT)) {
+                return parseLong(valueString);
+            }
+            if (type.equals(REAL)) {
+                return (long) floatToRawIntBits(parseFloat(valueString));
+            }
+            if (type.equals(DOUBLE)) {
+                return parseDouble(valueString);
+            }
+            if (type.equals(DATE)) {
+                return parseLong(valueString);
+            }
+            if (type instanceof VarcharType) {
+                return utf8Slice(valueString);
+            }
+            if (type.equals(VarbinaryType.VARBINARY)) {
+                return utf8Slice(valueString);
+            }
+            if (isShortDecimal(type) || isLongDecimal(type)) {
+                DecimalType decimalType = (DecimalType) type;
+                BigDecimal decimal = new BigDecimal(valueString);
+                decimal = decimal.setScale(decimalType.getScale(), BigDecimal.ROUND_UNNECESSARY);
+                if (decimal.precision() > decimalType.getPrecision()) {
+                    throw new IllegalArgumentException();
```

Comment:
Include the decimalPrecision and decimalType precision in the error message, along with the valueString.

Reply:
The same reaseon as above.